src/reduced_basis/rb_eim_construction.C

```diff
@@ -136,797 +111,831 @@ void RBEIMConstruction::set_best_fit_type_flag (const std::string & best_fit_typ
+      std::pair<Real,unsigned int> max_error_pair = compute_max_eim_error();
+      abs_greedy_error = max_error_pair.first;
+      current_training_index = max_error_pair.second;
+      set_params_from_training_set(current_training_index);
 
-void RBEIMConstruction::set_point_locator_tol(Real point_locator_tol)
-{
-  _point_locator_tol = point_locator_tol;
-}
+      libMesh::out << "Maximum EIM error is " << abs_greedy_error << std::endl << std::endl;
 
-Real RBEIMConstruction::get_point_locator_tol() const
-{
-  return _point_locator_tol;
+      // record the initial error
+      if (!initial_greedy_error_initialized)
+        {
+          initial_greedy_error = abs_greedy_error;
+          initial_greedy_error_initialized = true;
+        }
+
+      // Convergence and/or termination tests
+      {
+        if (rbe.get_n_basis_functions() >= this->get_Nmax())
+          {
+            libMesh::out << "Maximum number of basis functions reached: Nmax = "
+                          << get_Nmax() << std::endl;
+            break;
+          }
+
+        if (abs_greedy_error < get_abs_training_tolerance())
+          {
+            libMesh::out << "Absolute error tolerance reached." << std::endl;
+            break;
+          }
+
+        Real rel_greedy_error = abs_greedy_error/initial_greedy_error;
+        if (rel_greedy_error < get_rel_training_tolerance())
+          {
+            libMesh::out << "Relative error tolerance reached." << std::endl;
+            break;
+          }
+
+        if (rbe.get_n_basis_functions() >= this->get_Nmax())
+          {
+            libMesh::out << "Maximum number of basis functions reached: Nmax = "
+                         << get_Nmax() << std::endl;
+            break;
+          }
+
+        {
+          bool do_exit = false;
+          for (auto & param : greedy_param_list)
+            if (param == get_parameters())
+              {
+                libMesh::out << "Exiting greedy because the same parameters were selected twice"
+                             << std::endl;
+                do_exit = true;
+                break;
+              }
+
+          if (do_exit)
+            break; // out of while
+        }
+      }
+    } // end while(true)
+
+  return abs_greedy_error;
 }
 
 void RBEIMConstruction::initialize_eim_assembly_objects()
 {
   _rb_eim_assembly_objects.clear();
-  for (unsigned int i=0; i<get_rb_evaluation().get_n_basis_functions(); i++)
+  for (auto i : make_range(get_rb_eim_evaluation().get_n_basis_functions()))
     _rb_eim_assembly_objects.push_back(build_eim_assembly(i));
 }
 
-ExplicitSystem & RBEIMConstruction::get_explicit_system()
+std::vector<std::unique_ptr<ElemAssembly>> & RBEIMConstruction::get_eim_assembly_objects()
 {
-  return get_equation_systems().get_system<ExplicitSystem>(_explicit_system_name);
+  return _rb_eim_assembly_objects;
 }
 
-void RBEIMConstruction::load_basis_function(unsigned int i)
+void RBEIMConstruction::init_context(FEMContext & c)
```

Comment:
Ah, never mind.  My search just was turning up empty before because I hadn't yet hit "Load diff" on the big changes.